exdb/emails.py

```diff
@@ -0,0 +1,181 @@
+    """
+    Given a datatuple of (subject, text_content, html_content [optional], from_email,
+    recipient_list), sends each message to each recipient list. Returns the
+    number of emails sent.
+    If from_email is None, the DEFAULT_FROM_EMAIL setting is used.
+    If auth_user and auth_password are set, they're used to log in.
+    If auth_user is None, the EMAIL_HOST_USER setting is used.
+    If auth_password is None, the EMAIL_HOST_PASSWORD setting is used.
+    """
+    connection = connection or get_connection(
+        username=user, password=password, fail_silently=fail_silently
+    )
+
+    messages = []
+    for emailtuple in datatuple:
+        subject, text, html, from_email, recipient = emailtuple
+        message = EmailMultiAlternatives(subject, text, from_email, recipient, connection=connection)
+        message.attach_alternative(html, 'text/html')
+        messages.append(message)
+
+    return connection.send_messages(messages)
+
+
+class EmailTaskBase(object):
+    # task_name = "Email Task Base"
+
+    def send(self, emails, *args, **kwargs):  # pragma: no cover
+        raise NotImplementedError('send must be overridden for EmailTaskBase')
+
+
+class DailyDigest(EmailTaskBase):
+    """
+    This email should be sent to any user who is an approver daily
+    at 1600 local time.
+    """
+    task_name = "Daily Digest"
+
+    def is_time_to_send(self):
+        # Send the daily email at 1600
+        right_now = localtime(now())
+        return right_now.hour == 16 and (0 <= right_now.minute < 5)
+
+    def get_addrs(self):
+        """
+        Return all the users who have a pending experience they need
+        to approve, or an approval that needs evaluation
+        """
+        pending_experiences = Experience.objects.filter(status='pe')
+        experience_approvals_needing_eval = ExperienceApproval.objects.filter(
+            experience__status='ad', experience__end_datetime__lt=now())
+
+        users = get_user_model().objects.filter(
+            Q(approval_queue__pk__in=pending_experiences) |
+            Q(approval_set__pk__in=experience_approvals_needing_eval)
+        ).distinct()
+        return users
+
+    def send(self, *args, **kwargs):
+        if self.is_time_to_send():
+
+            emails = []
+            from_email = settings.SERVER_EMAIL
+            subject = settings.EMAIL_SUBJECT_PREFIX + ' Daily Digest'
+            for user in self.get_addrs():
+
+                # Find all the approvals that need evaluation and all the experiences that
+                # need to be approved by the current user
+                experience_approvals = ExperienceApproval.objects.filter(
+                    approver=user, experience__end_datetime__lt=now(), experience__status='ad')
+                experiences = Experience.objects.filter(Q(next_approver=user, status='pe') | Q(
+                    pk__in=experience_approvals.values('experience')))
+                experience_count = len(experiences)
+
```

Comment:
You should check that `experience_count > 0` before creating & appending the email. (We don't want to send off any blank emails.)